File: C++/1D_NF_mmult/FiringRate.cpp
#include "FiringRate.hpp"

FiringRate::FiringRate(double decay, double threshold){
    _miu = decay;
    _theta = threshold;
}


double FiringRate::f(double x){
    //return 1 / (1 + exp(- _miu * (u_i - _theta) ));
    return 1 / (1 + exp( -_miu*x + _theta )) - 1/(1 + exp(_theta));
}

void FiringRate::evaluateAt(const Vector& u, Vector& result){
    for (int i=0; i<result.GetSize(); i++){
        result[i] = f( u.Read(i) );
    }
}

File: Kernel_fRate Plots/1D/FiringRate.py
import numpy as np

class AbstractFiringRate:
	"""
	AbstractFiringRate Object
	-------------------
	Encapsulates the Parameters of the
	Firing Rate function for a Neural Field
	Integral equation of the Amari type.
	"""
	def __init__(self, decay, threshold):
		# Decay rate parameter.
		self.miu = decay
		# Threshold Parameter.
		self.theta = threshold

class FiringRate1(AbstractFiringRate):
	"""
	FiringRate1 Object
	-------------------
	Inherits data from AbstractFiringRate 
	"""
	def evaluateAt(self, u):
		"""
		Evaluate the Firing Rate Function at the
		given vector u.
		"""
		m = self.miu; h = self.theta
		return 1 / (1 + np.exp(- m * (u - h) ))

class FiringRate2(AbstractFiringRate):
	"""
	FiringRate2 Object
	-------------------
	Inherits data from AbstractFiringRate 
	"""
	def evaluateAt(self, u):
		"""
		Evaluate the Firing Rate Function at the
		given vector u.
		"""
		m = self.miu; h = self.theta
		return 1/( 1 + np.exp(- m*u + h) ) - 1/( 1 + np.exp(h) )

class laingFiringRate(AbstractFiringRate):
	"""
	laingFiringRate Object
	-------------------
	Firing Rate function as in Carlo R.Laing paper
	1D Model
	"""
	def evaluateAt(self, u):
		"""
		Evaluate the Firing Rate Function at the
		given vector u.
		"""
		m = self.miu; h = self.theta
		return 1/( 1 + np.exp(- m*u + h) ) - 1/( 1 + np.exp(h) )